src/odemis/util/peak.py

```diff
@@ -237,7 +274,8 @@ def _DoFit(self, future, spectrum, wavelength, type='gaussian'):
             logging.debug("Starting peak detection on data (len = %d) with window = %d",
                           len(wavelength), window_size)
             try:
-                width = PEAK_WIDTHS[type]
+                initial_width = PEAK_WIDTHS[type]
+                width = calc_width(initial_width, wavelength)
```

Comment:
I don't think it's useful to put it in a separate function. I'd find it more clear to directly see `width = wl_rng * WIDTH_RATIO`